service/src/test/java/bio/terra/workspace/pact/ProfileApiTest.java

```diff
@@ -0,0 +1,139 @@
+package bio.terra.workspace.pact;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import au.com.dius.pact.consumer.MockServer;
+import au.com.dius.pact.consumer.dsl.PactDslJsonBody;
+import au.com.dius.pact.consumer.dsl.PactDslWithProvider;
+import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
+import au.com.dius.pact.consumer.junit5.PactTestFor;
+import au.com.dius.pact.core.model.RequestResponsePact;
+import au.com.dius.pact.core.model.annotations.Pact;
+import bio.terra.workspace.app.configuration.external.SpendProfileConfiguration;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.spendprofile.SpendProfileId;
+import bio.terra.workspace.service.spendprofile.SpendProfileService;
+import bio.terra.workspace.service.spendprofile.exceptions.SpendUnauthorizedException;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import java.util.Optional;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+
+@Tag("pact-test")
+@ExtendWith(PactConsumerTestExt.class)
+public class ProfileApiTest {
+
+  static final String billingProfileId = "4a5afeaa-b3b2-fa51-8e4e-9dbf294b7837";
```

Comment:
Is there any reason to just not make this a random UUID? I feel like having a constant in the file makes it seem like the value is somehow important.

Reply:
In general, using random (changing) is bad practice with pact - it tends to give incorrect indications that the pact has changed.
It's easiest to have it as a static value so it can be used in both the pact mock setup, and the test itself.

I'll see if I can make it more clear what's going on, though.